qutip/core/operators.py

```diff
@@ -52,7 +54,56 @@
 from .dimensions import flatten
 
 
-def jmat(j, which=None):
+def qdiags(diagonals, offsets, dims=None, shape=None, *, dtype=_data.CSR):
+    """
+    Constructs an operator from an array of diagonals.
+
+    Parameters
+    ----------
+    diagonals : sequence of array_like
+        Array of elements to place along the selected diagonals.
+
+    offsets : sequence of ints
+        Sequence for diagonals to be set:
+            - k=0 main diagonal
+            - k>0 kth upper diagonal
+            - k<0 kth lower diagonal
+    dims : list, optional
+        Dimensions for operator
+
+    shape : list, tuple, optional
+        Shape of operator.  If omitted, a square operator large enough
+        to contain the diagonals is generated.
+
+    dtype : type or str
```

Comment:
In theory we should use an alias like numpy's `array_like` for these so we can just define what's valid in one place, but let's not worry about it right now.